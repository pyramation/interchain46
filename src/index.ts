export * from './codegen';

import { cosmos } from './codegen';

const {
    createGroupWithPolicy
} = cosmos.group.v1.MessageComposer.withTypeUrl;

const msg = createGroupWithPolicy({
    admin: 'a string',
    groupMetadata: 'string here',
    groupPolicyAsAdmin: true,
    groupPolicyMetadata: 'a string here too',
    members: [], // MemberRequest[],
    decisionPolicy: {
        typeUrl: '/cosmos.group.v1.PercentageDecisionPolicy',
        value: {

        }
    }

})