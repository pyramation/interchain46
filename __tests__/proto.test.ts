import { cosmos } from '../src';
import Long from 'long';

const { createGroupPolicy, createGroupWithPolicy } =
  cosmos.group.v1.MessageComposer.withTypeUrl;
const { PercentageDecisionPolicy } = cosmos.group.v1;

const msg = createGroupWithPolicy({
  admin: 'address',
  groupMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
  groupPolicyAsAdmin: true,
  groupPolicyMetadata: 'https://some.ipfs.w3s.link/group_metadata.json',
  members: [
    {
      address: 'address',
      weight: '10',
      metadata: 'First user'
    }
  ],
  decisionPolicy: {
    typeUrl: '/cosmos.group.v1.PercentageDecisionPolicy',
    value: PercentageDecisionPolicy.encode({
      percentage: '0.5',
      windows: {
        votingPeriod: {
          nanos: 0,
          seconds: Long.fromValue(100)
        },
        minExecutionPeriod: {
          nanos: 0,
          seconds: Long.fromValue(100)
        }
      }
    }).finish()
  }
});

it('works', () => {
  expect(msg).toMatchSnapshot();
});
