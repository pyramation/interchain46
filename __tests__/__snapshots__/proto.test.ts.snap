// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
{
  "typeUrl": "/cosmos.group.v1.MsgCreateGroupWithPolicy",
  "value": {
    "admin": "address",
    "decisionPolicy": {
      "typeUrl": "/cosmos.group.v1.PercentageDecisionPolicy",
      "value": {
        "data": [
          10,
          3,
          48,
          46,
          53,
          18,
          8,
          10,
          2,
          8,
          100,
          18,
          2,
          8,
          100,
        ],
        "type": "Buffer",
      },
    },
    "groupMetadata": "https://some.ipfs.w3s.link/group_metadata.json",
    "groupPolicyAsAdmin": true,
    "groupPolicyMetadata": "https://some.ipfs.w3s.link/group_metadata.json",
    "members": [
      {
        "address": "address",
        "metadata": "First user",
        "weight": "10",
      },
    ],
  },
}
`;
